Remove Eclipse project configuration files generated by m2e

HannesWell opened the pull request:
The removed files are generated by m2e if one imports the projects as maven projects into Eclipse. They are therefore unnecessary. Furthermore they sometimes change their content for different reasons and therefore show up as changed content in the git staging view.
Therefore I suggest to simply remove them.
@waynebeaton can you please have a look?

waynebeaton commented:
You've also removed the formatter configuration.

According to the log, I changed the JDT configuration files to tweak the formatter once a year ago and the next most recent change was in December 2020.

What problem are you trying to solve?

HannesWell commented:
> According to the log, I changed the JDT configuration files to tweak the formatter once a year ago and the next most recent change was in December 2020.

Then the settings should of course be kept. Reverted that part.



> What problem are you trying to solve?

When I import the projects from this repo I end up with git changes (due to changes in how m2e generates the .classpath file). Since this file is always generated instead of always commuting the changes in generated files I would simply remove them and let m2e generate them (which it does anyway).

![grafik](https://github.com/eclipse/dash-licenses/assets/44067969/95615af6-d061-4ad2-adf5-72e77d7bd0ab)

HannesWell commented:
@waynebeaton any objection to remove the .classpath and .project files, which are generated by m2e?
You will probably have to re-import you projects after you have rebased on this, but will in the future not need changes like in f390da5b24dbbfd906c499a4ee34588b6d1b3a44.

waynebeaton commented:
I'm old-school Eclipse. I may need a minute to think about this...

The JDT preferences file is actually there intentionally: it configures the projects with some specific formatting options that it applies automatically one save. It may be time to give that up as there is no special requirement to use an Eclipse IDE to work on this project and automatically apply formatting by those who do will almost certainly result in irrelevant changes.

HannesWell commented:
> The JDT preferences file is actually there intentionally: it configures the projects with some specific formatting options that it applies automatically one save. It may be time to give that up as there is no special requirement to use an Eclipse IDE to work on this project and automatically apply formatting by those who do will almost certainly result in irrelevant changes.

I restored the settings files and only remove the `.classpath` and `.project` files which just contain default data that are recreated by m2e during imports.
@waynebeaton are you fine with removing them? This probably means that, once this is merged you have to import the project that is already in your workspace once again because applying this commit then deletes the mentioned files.

waynebeaton commented:
I need to get over myself and just merge this. It's time to give up the formatter as well; along with the reasons that I've already cited, I have far too many formatter overrides in the content. 

I'll try to do that this week.
